src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1480,6 +1480,17 @@ class ActorIdStringGenerator : public Generator<std::string> {
     std::string _actorId;
 };
 
+/** `{^NowTimestamp: {}}` */
+class NowTimestampGenerator : public Generator<bsoncxx::types::b_timestamp> {
+public:
+    NowTimestampGenerator(const Node& node, GeneratorArgs generatorArgs) {}
+    bsoncxx::types::b_timestamp evaluate() override {
+        auto now = std::chrono::system_clock::now();
+        uint32_t nowSinceEpoch = std::chrono::duration_cast<std::chrono::seconds>(now.time_since_epoch()).count();
+        return bsoncxx::types::b_timestamp{.increment = 1, .timestamp = nowSinceEpoch};
```

Comment:
As per my understanding, bsoncxx::types::b_timestamp uses .increment to order (or compare) two timestamps in case the .timestamp is same.
For example,
if we have two documents with timestamp as
Doc 1 - Timestamp({ .timestamp: 1710804294, .increment: 1 }) 
Doc 2 - Timestamp({ .timestamp: 1710804294, .increment: 2 })
we use .increment to order these 2 documents.
For reference, [this](https://www.mongodb.com/docs/mongodb-shell/reference/data-types/#timestamp) is what I looked at.

I tried removing the .increment (as good as setting it to 0), the stream processing job did not pick any document from the collection.